### ISSUE TITLE
Option for shortcut

### ISSUE THREAD

**Issue by LawnSounds** [fnxweb/urllink#4]
I sometimes find it faster to just copy/paste words in a new tab for Googling something so i was wondering if you could implement an option for shortcuts? Eg. a shortcut for a list of the options or a shortcut for each option, eg. opening in a new tab unaltered and so on?

**Comment by fnxweb** [comment]
Yes, I had to drop keyboard shortcuts when I converted it to a WebExtension as it was about thing that's completely different now.

I seem to recall that they can be added, but have to be hard-coded, and I was worried about choosing something that clashes for people, and them but being able to change it.

Made more tricky by the different OS releases if Firefox have different built-in shortcuts as well!

Although I may be able to be a bit clever about the shortcut assignments...  I'll have a play.

One thing that won't work yet is menu-option “letter” shortcuts (where the shortcut once the menu is up has an underscore under it).  That is (still) in, but hasn't been implemented in Firefox yet.

When they've put the feature back in, I'll re-enable that in URL Link.

**Comment by fnxweb** [comment]
Wow, keyboard shortcut support for WebExtensions is a complete car-crash at the moment.  I'll have a play, though, and see what can be done.

**Comment by LawnSounds** [comment]
The Emoji helper (Also called Emoji Cheatsheet) has a shortcut, although not customisable, here: https://github.com/johannhof/emoji-helper

**Comment by fnxweb** [comment]
Yeah, you can currently only add shortcuts to predefined functions (so I may be able to add one to pop up the submenu), but there's no API to change that shortcut (and if it clashes with a browser one, it's ignored), not is there a way to add further ones (so you can't have a shortcut for a custom URL Link menu entry).  _(e.g., see the **manifest.json** file for the Emoji helper where it sets <Ctrl-Shift-E> — but that's for a popup not a context menu)_

_sigh_

I'm following the Firefox bugs that are requesting a shortcut API, so that I know when I can do something.  In the short-term, I'll see if one of the things that can be done with a shortcut is pop up the context menu options.
